In [66]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [67]:
# Loading the datasets
prep_df = pd.read_csv('climate_precip.csv')
temp_df = pd.read_csv('climate_temp.csv')

In [68]:
print(f'Shape of the precipitaion data: {prep_df.shape}')
print(f'Shape of the temperature data: {temp_df.shape}')

Shape of the precipitaion data: (151110, 29)
Shape of the temperature data: (127020, 21)


In [69]:
pd.set_option('display.max_columns', 100)

In [70]:
# Overview of the datasets 
# prep_df.sample(5)
prep_df.head(5)

STATION            STATION_NAME      DATE  DLY-PRCP-25PCTL  \
0  GHCND:USC00049099  TWENTYNINE PALMS CA US  20100101            -6.66   
1  GHCND:USC00049099  TWENTYNINE PALMS CA US  20100102            -6.66   
2  GHCND:USC00049099  TWENTYNINE PALMS CA US  20100103            -6.66   
3  GHCND:USC00049099  TWENTYNINE PALMS CA US  20100104            -6.66   
4  GHCND:USC00049099  TWENTYNINE PALMS CA US  20100105            -6.66   

   DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  DLY-SNWD-50PCTL  \
0             -666            -66.6            -6.66             -666   
1             -666            -66.6            -6.66             -666   
2             -666            -66.6            -6.66             -666   
3            -9999          -9999.0            -6.66            -9999   
4            -9999          -9999.0            -6.66            -9999   

   DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  \
0            -66.6            -6.66             -666            -66.6   
1            -66.6            -6.66             -666            -66.6   
2            -66.6            -6.66             -666            -66.6   
3          -9999.0            -6.66            -9999          -9999.0   
4          -9999.0            -6.66            -9999          -9999.0   

   MTD-PRCP-NORMAL  MTD-SNOW-NORMAL  YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  \
0             0.02              0.0             0.02              0.0   
1             0.04              0.0             0.04              0.0   
2             0.05              0.0             0.05              0.0   
3             0.07              0.0             0.07              0.0   
4             0.09              0.0             0.09              0.0   

   DLY-PRCP-PCTALL-GE001HI  DLY-PRCP-PCTALL-GE010HI  DLY-PRCP-PCTALL-GE050HI  \
0                       98                       43                       12   
1                       99                       44                       12   
2                      100                       44                       12   
3                      101                       45                       12   
4                      102                       45                       12   

   DLY-PRCP-PCTALL-GE100HI  DLY-SNWD-PCTALL-GE001WI  DLY-SNWD-PCTALL-GE010WI  \
0                        3                    -9999                        0   
1                        3                    -9999                        0   
2                        3                    -9999                        0   
3                        3                        0                        0   
4                        3                        0                        0   

   DLY-SNWD-PCTALL-GE003WI  DLY-SNWD-PCTALL-GE005WI  DLY-SNOW-PCTALL-GE001TI  \
0                    -9999                    -9999                    -9999   
1                    -9999                    -9999                    -9999   
2                    -9999                    -9999                    -9999   
3                        0                        0                        0   
4                        0                        0                        0   

   DLY-SNOW-PCTALL-GE010TI  DLY-SNOW-PCTALL-GE100TI  DLY-SNOW-PCTALL-GE030TI  \
0                    -9999                        0                    -9999   
1                    -9999                        0                    -9999   
2                    -9999                        0                    -9999   
3                        0                        0                        0   
4                        0                        0                        0   

   DLY-SNOW-PCTALL-GE050TI  
0                    -9999  
1                    -9999  
2                    -9999  
3                        0  
4                        0

In [71]:
temp_df.head()

STATION            STATION_NAME ELEVATION  LATITUDE   LONGITUDE  \
0  GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806  -116.03694   
1  GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806  -116.03694   
2  GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806  -116.03694   
3  GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806  -116.03694   
4  GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806  -116.03694   

       DATE  DLY-CLDD-BASE45  DLY-CLDD-BASE50  DLY-CLDD-BASE55  \
0  20100101                6                2            -7777   
1  20100102                6                2                1   
2  20100103                6                2                1   
3  20100104                6                2                1   
4  20100105                6                2                1   

   DLY-CLDD-BASE57  DLY-CLDD-BASE60  DLY-CLDD-NORMAL  DLY-CLDD-BASE70  \
0            -7777            -7777                0                0   
1            -7777            -7777                0                0   
2            -7777            -7777                0                0   
3            -7777            -7777                0                0   
4            -7777            -7777                0                0   

   DLY-CLDD-BASE72  DLY-HTDD-BASE40  DLY-HTDD-BASE45  DLY-HTDD-BASE50  \
0                0            -7777                1                2   
1                0            -7777                1                2   
2                0            -7777                1                2   
3                0            -7777                1                2   
4                0            -7777            -7777                2   

   DLY-HTDD-BASE55  DLY-HTDD-BASE57  DLY-HTDD-BASE60  DLY-HTDD-NORMAL  
0                6                7               10               15  
1                6                7               10               15  
2                5                7               10               15  
3                5                7               10               15  
4                5                7               10               15

In [14]:
# GEtting the info about the dataset and their columns
print('Info of the precipitaion dataset:')
print(prep_df.info())
# print('Info of the temperature dataset:')
# print(temp_df.info())

Info of the precipitaion dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 151110 entries, 0 to 151109
Data columns (total 29 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   STATION                  151110 non-null  object 
 1   STATION_NAME             151110 non-null  object 
 2   DATE                     151110 non-null  int64  
 3   DLY-PRCP-25PCTL          151110 non-null  float64
 4   DLY-SNWD-25PCTL          151110 non-null  int64  
 5   DLY-SNOW-25PCTL          151110 non-null  float64
 6   DLY-PRCP-50PCTL          151110 non-null  float64
 7   DLY-SNWD-50PCTL          151110 non-null  int64  
 8   DLY-SNOW-50PCTL          151110 non-null  float64
 9   DLY-PRCP-75PCTL          151110 non-null  float64
 10  DLY-SNWD-75PCTL          151110 non-null  int64  
 11  DLY-SNOW-75PCTL          151110 non-null  float64
 12  MTD-PRCP-NORMAL          151110 non-null  float64
 13  MTD-SNOW-NORMAL          

In [16]:
# Statistical summary of each dataset
prep_df.describe()

DATE  DLY-PRCP-25PCTL  DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  \
count  1.511100e+05    151110.000000    151110.000000    151110.000000   
mean   2.010067e+07     -1983.400927     -9099.834584     -8477.923237   
std    3.450030e+02      3984.261185      2786.079371      3579.071334   
min    2.010010e+07     -9999.000000     -9999.000000     -9999.000000   
25%    2.010040e+07        -6.660000     -9999.000000     -9999.000000   
50%    2.010070e+07        -6.660000     -9999.000000     -9999.000000   
75%    2.010100e+07         0.060000     -9999.000000     -9999.000000   
max    2.010123e+07         0.290000        40.000000         2.000000   

       DLY-PRCP-50PCTL  DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
count    151110.000000    151110.000000    151110.000000    151110.000000   
mean      -1983.332954     -9099.703084     -8477.890501     -1983.185442   
std        3984.295025      2786.509451      3579.148894      3984.368470   
min       -9999.000000     -9999.000000     -9999.000000     -9999.000000   
25%          -6.660000     -9999.000000     -9999.000000        -6.660000   
50%          -6.660000     -9999.000000     -9999.000000        -6.660000   
75%           0.180000     -9999.000000     -9999.000000         0.460000   
max           0.770000        55.000000         4.100000         1.800000   

       DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  MTD-SNOW-NORMAL  \
count    151110.000000    151110.000000    151110.000000    151110.000000   
mean      -9099.506492     -8477.832838         1.019360     -2849.609671   
std        2787.152875      3579.285544         1.559976      4514.042489   
min       -9999.000000     -9999.000000         0.000000     -9999.000000   
25%       -9999.000000     -9999.000000         0.080000     -9999.000000   
50%       -9999.000000     -9999.000000         0.380000         0.000000   
75%       -9999.000000     -9999.000000         1.290000         0.000000   
max          72.000000         9.200000        17.620000        51.500000   

       YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  DLY-PRCP-PCTALL-GE001HI  \
count    151110.000000    151110.000000            151110.000000   
mean         13.704885     -2844.786476             -1871.283628   
std          10.798821      4517.128935              4060.994764   
min           0.010000     -9999.000000             -9999.000000   
25%           6.160000     -9999.000000                11.000000   
50%          11.030000         0.000000                75.000000   
75%          18.290000         0.100000               212.000000   
max          90.730000       218.200000               615.000000   

       DLY-PRCP-PCTALL-GE010HI  DLY-PRCP-PCTALL-GE050HI  \
count            151110.000000            151110.000000   
mean              -2071.147429             -2208.350572   
std                4154.288671              4188.374424   
min               -9999.000000             -9999.000000   
25%                   3.000000                 0.000000   
50%                  40.000000                10.000000   
75%                 139.000000                48.000000   
max                 470.000000               323.000000   

       DLY-PRCP-PCTALL-GE100HI  DLY-SNWD-PCTALL-GE001WI  \
count            151110.000000            151110.000000   
mean              -2335.718463             -3384.903891   
std                4247.919384              4745.858699   
min               -9999.000000             -9999.000000   
25%                   0.000000             -9999.000000   
50%                   2.000000                 0.000000   
75%                  16.000000                 0.000000   
max                 205.000000              1000.000000   

       DLY-SNWD-PCTALL-GE010WI  DLY-SNWD-PCTALL-GE003WI  \
count            151110.000000            151110.000000   
mean              -3315.576196             -3346.827655   
std                4714.756621              4730.176154   
min               -9999.000000             -999

In [17]:
temp_df.describe()

DATE  DLY-CLDD-BASE45  DLY-CLDD-BASE50  DLY-CLDD-BASE55  \
count  1.270200e+05    127020.000000    127020.000000    127020.000000   
mean   2.010067e+07      -305.681680      -571.357164     -1364.776594   
std    3.450032e+02      1550.917511      2050.944179      2968.682933   
min    2.010010e+07     -7777.000000     -7777.000000     -7777.000000   
25%    2.010040e+07         7.000000         3.000000         1.000000   
50%    2.010070e+07        14.000000         9.000000         5.000000   
75%    2.010100e+07        23.000000        18.000000        13.000000   
max    2.010123e+07        58.000000        53.000000        48.000000   

       DLY-CLDD-BASE57  DLY-CLDD-BASE60  DLY-CLDD-NORMAL  DLY-CLDD-BASE70  \
count    127020.000000    127020.000000    127020.000000    127020.000000   
mean      -1610.030515     -1993.233766     -2300.983097     -2409.871020   
std        3159.342811      3401.256167      3553.088077      3598.276966   
min       -7777.000000     -7777.000000     -7777.000000     -7777.000000   
25%           0.000000     -7777.000000     -7777.000000     -7777.000000   
50%           3.000000         2.000000         0.000000         0.000000   
75%          11.000000         8.000000         4.000000         1.000000   
max          46.000000        43.000000        38.000000        33.000000   

       DLY-CLDD-BASE72  DLY-HTDD-BASE40  DLY-HTDD-BASE45  DLY-HTDD-BASE50  \
count    127020.000000    127020.000000    127020.000000    127020.000000   
mean      -2355.161754     -1797.129413     -2160.404653     -2134.298292   
std        3574.876074      3278.714446      3484.418347      3472.336747   
min       -7777.000000     -7777.000000     -7777.000000     -7777.000000   
25%       -7777.000000         0.000000     -7777.000000     -7777.000000   
50%           0.000000         0.000000         0.000000         0.000000   
75%           1.000000         0.000000         0.000000         2.000000   
max          31.000000        18.000000        23.000000        28.000000   

       DLY-HTDD-BASE55  DLY-HTDD-BASE57  DLY-HTDD-BASE60  DLY-HTDD-NORMAL  
count    127020.000000    127020.000000    127020.000000    127020.000000  
mean      -1867.945599     -1801.272060     -1697.664328     -1214.191041  
std        3326.150925      3285.663313      3219.225586      2834.496038  
min       -7777.000000     -7777.000000     -7777.000000     -7777.000000  
25%           0.000000         0.000000         0.000000         1.000000  
50%           1.000000         1.000000         3.000000         6.000000  
75%           4.000000         6.000000         9.000000        14.000000  
max          33.000000        35.000000        38.000000        43.000000

In [18]:
# Checking is there any missing values in the datasets
prep_df.isnull().sum()

STATION                    0
STATION_NAME               0
DATE                       0
DLY-PRCP-25PCTL            0
DLY-SNWD-25PCTL            0
DLY-SNOW-25PCTL            0
DLY-PRCP-50PCTL            0
DLY-SNWD-50PCTL            0
DLY-SNOW-50PCTL            0
DLY-PRCP-75PCTL            0
DLY-SNWD-75PCTL            0
DLY-SNOW-75PCTL            0
MTD-PRCP-NORMAL            0
MTD-SNOW-NORMAL            0
YTD-PRCP-NORMAL            0
YTD-SNOW-NORMAL            0
DLY-PRCP-PCTALL-GE001HI    0
DLY-PRCP-PCTALL-GE010HI    0
DLY-PRCP-PCTALL-GE050HI    0
DLY-PRCP-PCTALL-GE100HI    0
DLY-SNWD-PCTALL-GE001WI    0
DLY-SNWD-PCTALL-GE010WI    0
DLY-SNWD-PCTALL-GE003WI    0
DLY-SNWD-PCTALL-GE005WI    0
DLY-SNOW-PCTALL-GE001TI    0
DLY-SNOW-PCTALL-GE010TI    0
DLY-SNOW-PCTALL-GE100TI    0
DLY-SNOW-PCTALL-GE030TI    0
DLY-SNOW-PCTALL-GE050TI    0
dtype: int64

In [19]:
temp_df.isnull().sum()

STATION            0
STATION_NAME       0
ELEVATION          0
LATITUDE           0
LONGITUDE          0
DATE               0
DLY-CLDD-BASE45    0
DLY-CLDD-BASE50    0
DLY-CLDD-BASE55    0
DLY-CLDD-BASE57    0
DLY-CLDD-BASE60    0
DLY-CLDD-NORMAL    0
DLY-CLDD-BASE70    0
DLY-CLDD-BASE72    0
DLY-HTDD-BASE40    0
DLY-HTDD-BASE45    0
DLY-HTDD-BASE50    0
DLY-HTDD-BASE55    0
DLY-HTDD-BASE57    0
DLY-HTDD-BASE60    0
DLY-HTDD-NORMAL    0
dtype: int64

In [23]:
x = prep_df.loc[prep_df['STATION'] == 'GHCND:USC00045721', :]
x # subset of the data

STATION            STATION_NAME      DATE  DLY-PRCP-25PCTL  \
1460  GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101             0.04   
1461  GHCND:USC00045721  MITCHELL CAVERNS CA US  20100102             0.05   
1462  GHCND:USC00045721  MITCHELL CAVERNS CA US  20100103             0.05   
1463  GHCND:USC00045721  MITCHELL CAVERNS CA US  20100104             0.05   
1464  GHCND:USC00045721  MITCHELL CAVERNS CA US  20100105             0.05   
...                 ...                     ...       ...              ...   
1820  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101227             0.04   
1821  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101228             0.04   
1822  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101229             0.04   
1823  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101230             0.04   
1824  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231             0.04   

      DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  DLY-SNWD-50PCTL  \
1460             -666            -66.6             0.16             -666   
1461             -666            -66.6             0.16             -666   
1462             -666            -66.6             0.16             -666   
1463             -666            -66.6             0.16             -666   
1464             -666            -66.6             0.17             -666   
...               ...              ...              ...              ...   
1820             -666            -66.6             0.15             -666   
1821             -666            -66.6             0.15             -666   
1822             -666            -66.6             0.15             -666   
1823             -666            -66.6             0.15             -666   
1824             -666            -66.6             0.15             -666   

      DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  \
1460            -66.6             0.44             -666            -66.6   
1461            -66.6             0.44             -666            -66.6   
1462            -66.6             0.45             -666            -66.6   
1463            -66.6             0.45             -666            -66.6   
1464            -66.6             0.46             -666            -66.6   
...               ...              ...              ...              ...   
1820            -66.6             0.44             -666            -66.6   
1821            -66.6             0.43             -666            -66.6   
1822            -66.6             0.43             -666            -66.6   
1823            -66.6             0.43             -666            -66.6   
1824            -66.6             0.44             -666            -66.6   

      MTD-PRCP-NORMAL  MTD-SNOW-NORMAL  YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  \
1460             0.04              0.0             0.04              0.0   
1461             0.08              0.0             0.08              0.0   
1462             0.12              0.0             0.12              0.0   
1463             0.16              0.0             0.16              0.0   
1464             0.21              0.0             0.21              0.0   
...               ...              ...              ...              ...   
1820             1.14              0.3            11.29              1.5   
1821             1.19              0.3            11.34              1.5   
1822             1.24              0.3            11.39              1.5   
1823             1.30              0.3            11.45              1.5   
1824             1.35              0.3            11.50              1.5   

      DLY-PRCP-PCTALL-GE001HI  DLY-PRCP-PCTALL-GE010HI  \
1460                      137                       81   
1461                      138                       83   
1462                      139                       84   
1463                      140                       85   
1464                      141                     

In [51]:
x.STATION.unique(), x.STATION_NAME.unique()

(array(['GHCND:USC00045721'], dtype=object),
 array(['MITCHELL CAVERNS CA US'], dtype=object))

In [52]:
pd.merge(prep_df, x, on='STATION', how='inner')

STATION          STATION_NAME_x    DATE_x  \
0       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101   
1       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101   
2       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101   
3       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101   
4       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101   
...                   ...                     ...       ...   
133220  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231   
133221  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231   
133222  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231   
133223  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231   
133224  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231   

        DLY-PRCP-25PCTL_x  DLY-SNWD-25PCTL_x  DLY-SNOW-25PCTL_x  \
0                    0.04               -666              -66.6   
1                    0.04               -666              -66.6   
2                    0.04               -666              -66.6   
3                    0.04               -666              -66.6   
4                    0.04               -666              -66.6   
...                   ...                ...                ...   
133220               0.04               -666              -66.6   
133221               0.04               -666              -66.6   
133222               0.04               -666              -66.6   
133223               0.04               -666              -66.6   
133224               0.04               -666              -66.6   

        DLY-PRCP-50PCTL_x  DLY-SNWD-50PCTL_x  DLY-SNOW-50PCTL_x  \
0                    0.16               -666              -66.6   
1                    0.16               -666              -66.6   
2                    0.16               -666              -66.6   
3                    0.16               -666              -66.6   
4                    0.16               -666              -66.6   
...                   ...                ...                ...   
133220               0.15               -666              -66.6   
133221               0.15               -666              -66.6   
133222               0.15               -666              -66.6   
133223               0.15               -666              -66.6   
133224               0.15               -666              -66.6   

        DLY-PRCP-75PCTL_x  DLY-SNWD-75PCTL_x  DLY-SNOW-75PCTL_x  \
0                    0.44               -666              -66.6   
1                    0.44               -666              -66.6   
2                    0.44               -666              -66.6   
3                    0.44               -666              -66.6   
4                    0.44               -666              -66.6   
...                   ...                ...                ...   
133220               0.44               -666              -66.6   
133221               0.44               -666              -66.6   
133222               0.44               -666              -66.6   
133223               0.44               -666              -66.6   
133224               0.44               -666              -66.6   

        MTD-PRCP-NORMAL_x  MTD-SNOW-NORMAL_x  YTD-PRCP-NORMAL_x  \
0                    0.04                0.0               0.04   
1                    0.04                0.0               0.04   
2                    0.04                0.0               0.04   
3                    0.04                0.0               0.04   
4                    0.04                0.0               0.04   
...                   ...                ...                ...   
133220               1.35                0.3              11.50   
133221               1.35                0.3              11.50   
133222               1.35                0.3              11.50   
133223               1.35                0.3              11.50   
133224               1.35                0.3              11.50   

        YTD-SNOW-NORMAL_x  DLY-PRCP-PCTAL

In [25]:
pd.merge(prep_df, x, on='DATE', how='inner')

STATION_x          STATION_NAME_x      DATE  \
0       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100101   
1       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100102   
2       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100103   
3       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100104   
4       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100105   
...                   ...                     ...       ...   
151105  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101227   
151106  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101228   
151107  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101229   
151108  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101230   
151109  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101231   

        DLY-PRCP-25PCTL_x  DLY-SNWD-25PCTL_x  DLY-SNOW-25PCTL_x  \
0                   -6.66               -666              -66.6   
1                   -6.66               -666              -66.6   
2                   -6.66               -666              -66.6   
3                   -6.66              -9999            -9999.0   
4                   -6.66              -9999            -9999.0   
...                   ...                ...                ...   
151105               0.12              -9999            -9999.0   
151106               0.12              -9999            -9999.0   
151107               0.12              -9999            -9999.0   
151108               0.12              -9999            -9999.0   
151109               0.12              -9999            -9999.0   

        DLY-PRCP-50PCTL_x  DLY-SNWD-50PCTL_x  DLY-SNOW-50PCTL_x  \
0                   -6.66               -666              -66.6   
1                   -6.66               -666              -66.6   
2                   -6.66               -666              -66.6   
3                   -6.66              -9999            -9999.0   
4                   -6.66              -9999            -9999.0   
...                   ...                ...                ...   
151105               0.41              -9999            -9999.0   
151106               0.42              -9999            -9999.0   
151107               0.43              -9999            -9999.0   
151108               0.43              -9999            -9999.0   
151109               0.44              -9999            -9999.0   

        DLY-PRCP-75PCTL_x  DLY-SNWD-75PCTL_x  DLY-SNOW-75PCTL_x  \
0                   -6.66               -666              -66.6   
1                   -6.66               -666              -66.6   
2                   -6.66               -666              -66.6   
3                   -6.66              -9999            -9999.0   
4                   -6.66              -9999            -9999.0   
...                   ...                ...                ...   
151105               1.13              -9999            -9999.0   
151106               1.14              -9999            -9999.0   
151107               1.15              -9999            -9999.0   
151108               1.15              -9999            -9999.0   
151109               1.16              -9999            -9999.0   

        MTD-PRCP-NORMAL_x  MTD-SNOW-NORMAL_x  YTD-PRCP-NORMAL_x  \
0                    0.02                0.0               0.02   
1                    0.04                0.0               0.04   
2                    0.05                0.0               0.05   
3                    0.07                0.0               0.07   
4                    0.09                0.0               0.09   
...                   ...                ...                ...   
151105               3.91            -9999.0              35.20   
151106               4.10            -9999.0              35.39   
151107               4.30            -9999.0              35.59   
151108               4.50            -9999.0              35.79   
151109               4.69            -9999.0              35.98   

        YTD-SNOW-NORMAL_x  DLY-PRCP-PCT

##### 29 columns from precipitation data, 29 columns from subset and DATE column is same so 29 + 29 - 1 = 57
##### And getting all 151110 rows because we are merging on DATE, and each day all the stations share climate report ...

In [28]:
prep_df.shape 

(151110, 29)

In [29]:
# Merging entire datasets
pd.merge(prep_df, temp_df, on=['DATE', 'STATION'], how='inner')

STATION          STATION_NAME_x      DATE  DLY-PRCP-25PCTL  \
0       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100101            -6.66   
1       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100102            -6.66   
2       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100103            -6.66   
3       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100104            -6.66   
4       GHCND:USC00049099  TWENTYNINE PALMS CA US  20100105            -6.66   
...                   ...                     ...       ...              ...   
123000  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101227             0.12   
123001  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101228             0.12   
123002  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101229             0.12   
123003  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101230             0.12   
123004  GHCND:USC00046006  MOUNT WILSON CBS CA US  20101231             0.12   

        DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  DLY-SNWD-50PCTL  \
0                  -666            -66.6            -6.66             -666   
1                  -666            -66.6            -6.66             -666   
2                  -666            -66.6            -6.66             -666   
3                 -9999          -9999.0            -6.66            -9999   
4                 -9999          -9999.0            -6.66            -9999   
...                 ...              ...              ...              ...   
123000            -9999          -9999.0             0.41            -9999   
123001            -9999          -9999.0             0.42            -9999   
123002            -9999          -9999.0             0.43            -9999   
123003            -9999          -9999.0             0.43            -9999   
123004            -9999          -9999.0             0.44            -9999   

        DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  \
0                 -66.6            -6.66             -666            -66.6   
1                 -66.6            -6.66             -666            -66.6   
2                 -66.6            -6.66             -666            -66.6   
3               -9999.0            -6.66            -9999          -9999.0   
4               -9999.0            -6.66            -9999          -9999.0   
...                 ...              ...              ...              ...   
123000          -9999.0             1.13            -9999          -9999.0   
123001          -9999.0             1.14            -9999          -9999.0   
123002          -9999.0             1.15            -9999          -9999.0   
123003          -9999.0             1.15            -9999          -9999.0   
123004          -9999.0             1.16            -9999          -9999.0   

        MTD-PRCP-NORMAL  MTD-SNOW-NORMAL  YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  \
0                  0.02              0.0             0.02              0.0   
1                  0.04              0.0             0.04              0.0   
2                  0.05              0.0             0.05              0.0   
3                  0.07              0.0             0.07              0.0   
4                  0.09              0.0             0.09              0.0   
...                 ...              ...              ...              ...   
123000             3.91          -9999.0            35.20          -9999.0   
123001             4.10          -9999.0            35.39          -9999.0   
123002             4.30          -9999.0            35.59          -9999.0   
123003             4.50          -9999.0            35.79          -9999.0   
123004             4.69          -9999.0            35.98          -9999.0   

        DLY-PRCP-PCTALL-GE001HI  DLY-PRCP-PCTALL-GE010HI  \
0                            98                       43   
1                            99                       44   
2                           100                       44   
3     

In [38]:
print(len(prep_df.columns))
print(len(temp_df.columns))

29
21


In [44]:
# Outer join 
pd.merge(temp_df, x, on='DATE', how='outer')

STATION_x                              STATION_NAME_x  \
0       GHCND:USC00049099                      TWENTYNINE PALMS CA US   
1       GHCND:USC00041244                          BUTTONWILLOW CA US   
2       GHCND:USC00047902                         SANTA BARBARA CA US   
3       GHCND:USC00042214                           CULVER CITY CA US   
4       GHCND:USC00045721                      MITCHELL CAVERNS CA US   
...                   ...                                         ...   
127015  GHCND:USC00041912                                COLFAX CA US   
127016  GHCND:USC00047339                          REDWOOD CITY CA US   
127017  GHCND:USC00040943  BODIE CALIFORNIA STATE HISTORIC PARK CA US   
127018  GHCND:USC00045032                                  LODI CA US   
127019  GHCND:USC00046006                      MOUNT WILSON CBS CA US   

       ELEVATION  LATITUDE   LONGITUDE      DATE  DLY-CLDD-BASE45  \
0            602  34.12806  -116.03694  20100101                6   
1             82   35.4047   -119.4731  20100101                2   
2            4.9   34.4167   -119.6844  20100101               10   
3           16.8    34.005   -118.4127  20100101               11   
4         1325.9   34.9436   -115.5469  20100101                3   
...          ...       ...         ...       ...              ...   
127015     725.4   39.0911    -120.948  20101231                1   
127016       9.4  37.47667  -122.23861  20101231                4   
127017    2551.2   38.2119   -119.0142  20101231                0   
127018      12.2  38.10611  -121.28778  20101231                3   
127019    1740.4   34.2308   -118.0711  20101231                4   

        DLY-CLDD-BASE50  DLY-CLDD-BASE55  DLY-CLDD-BASE57  DLY-CLDD-BASE60  \
0                     2            -7777            -7777            -7777   
1                     1            -7777            -7777            -7777   
2                     6                2                1            -7777   
3                     6                2                1            -7777   
4                     1            -7777            -7777            -7777   
...                 ...              ...              ...              ...   
127015            -7777                0                0                0   
127016                1            -7777            -7777            -7777   
127017                0                0                0                0   
127018                1            -7777            -7777            -7777   
127019                2                1            -7777            -7777   

        DLY-CLDD-NORMAL  DLY-CLDD-BASE70  DLY-CLDD-BASE72  DLY-HTDD-BASE40  \
0                     0                0                0            -7777   
1                     0                0                0                1   
2                 -7777                0                0                0   
3                 -7777                0                0                0   
4                     0                0                0                1   
...                 ...              ...              ...              ...   
127015                0                0                0                1   
127016                0                0                0            -7777   
127017                0                0                0               18   
127018                0                0                0            -7777   
127019            -7777                0                0                1   

        DLY-HTDD-BASE45  DLY-HTDD-BASE50  DLY-HTDD-BASE55  DLY-HTDD-BASE57  \
0                     1                2                6                7   
1                     2                6               10               12   
2                 -7777            -7777                2                3   
3                     0            -7777                1                2   
4                     3                6      

In [46]:
x.shape, temp_df.shape

((365, 29), (127020, 21))

In [54]:
# Left join of climate_temp data with subset x
pd.merge(left=temp_df, right=x, on='STATION', how='left')

STATION          STATION_NAME_x ELEVATION  LATITUDE  \
0       GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
1       GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
2       GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
3       GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
4       GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
...                   ...                     ...       ...       ...   
259875  GHCND:USC00046006  MOUNT WILSON CBS CA US    1740.4   34.2308   
259876  GHCND:USC00046006  MOUNT WILSON CBS CA US    1740.4   34.2308   
259877  GHCND:USC00046006  MOUNT WILSON CBS CA US    1740.4   34.2308   
259878  GHCND:USC00046006  MOUNT WILSON CBS CA US    1740.4   34.2308   
259879  GHCND:USC00046006  MOUNT WILSON CBS CA US    1740.4   34.2308   

         LONGITUDE    DATE_x  DLY-CLDD-BASE45  DLY-CLDD-BASE50  \
0       -116.03694  20100101                6                2   
1       -116.03694  20100102                6                2   
2       -116.03694  20100103                6                2   
3       -116.03694  20100104                6                2   
4       -116.03694  20100105                6                2   
...            ...       ...              ...              ...   
259875   -118.0711  20101227                4                2   
259876   -118.0711  20101228                4                2   
259877   -118.0711  20101229                4                2   
259878   -118.0711  20101230                4                2   
259879   -118.0711  20101231                4                2   

        DLY-CLDD-BASE55  DLY-CLDD-BASE57  DLY-CLDD-BASE60  DLY-CLDD-NORMAL  \
0                 -7777            -7777            -7777                0   
1                     1            -7777            -7777                0   
2                     1            -7777            -7777                0   
3                     1            -7777            -7777                0   
4                     1            -7777            -7777                0   
...                 ...              ...              ...              ...   
259875                1            -7777            -7777            -7777   
259876                1            -7777            -7777            -7777   
259877                1            -7777            -7777            -7777   
259878                1            -7777            -7777            -7777   
259879                1            -7777            -7777            -7777   

        DLY-CLDD-BASE70  DLY-CLDD-BASE72  DLY-HTDD-BASE40  DLY-HTDD-BASE45  \
0                     0                0            -7777                1   
1                     0                0            -7777                1   
2                     0                0            -7777                1   
3                     0                0            -7777                1   
4                     0                0            -7777            -7777   
...                 ...              ...              ...              ...   
259875                0                0                2                4   
259876                0                0                2                3   
259877                0                0                2                3   
259878                0                0                2                3   
259879                0                0                1                3   

        DLY-HTDD-BASE50  DLY-HTDD-BASE55  DLY-HTDD-BASE57  DLY-HTDD-BASE60  \
0                     2                6                7               10   
1                     2                6                7               10   
2                     2                5                7               10   
3                     2                5                7               10   
4                     2                5                7               10   
...   

In [55]:
# verifying the shape of the datasets
prep_df.shape, temp_df.shape, x.shape

((151110, 29), (127020, 21), (365, 29))

In [57]:
# Right join of climate_temp data with subset x

pd.merge(left = x, right=temp_df, on='DATE', how='right')

STATION_x          STATION_NAME_x      DATE  DLY-PRCP-25PCTL  \
0       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100101             0.04   
1       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100102             0.05   
2       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100103             0.05   
3       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100104             0.05   
4       GHCND:USC00045721  MITCHELL CAVERNS CA US  20100105             0.05   
...                   ...                     ...       ...              ...   
127015  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101227             0.04   
127016  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101228             0.04   
127017  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101229             0.04   
127018  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101230             0.04   
127019  GHCND:USC00045721  MITCHELL CAVERNS CA US  20101231             0.04   

        DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  DLY-SNWD-50PCTL  \
0                  -666            -66.6             0.16             -666   
1                  -666            -66.6             0.16             -666   
2                  -666            -66.6             0.16             -666   
3                  -666            -66.6             0.16             -666   
4                  -666            -66.6             0.17             -666   
...                 ...              ...              ...              ...   
127015             -666            -66.6             0.15             -666   
127016             -666            -66.6             0.15             -666   
127017             -666            -66.6             0.15             -666   
127018             -666            -66.6             0.15             -666   
127019             -666            -66.6             0.15             -666   

        DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  \
0                 -66.6             0.44             -666            -66.6   
1                 -66.6             0.44             -666            -66.6   
2                 -66.6             0.45             -666            -66.6   
3                 -66.6             0.45             -666            -66.6   
4                 -66.6             0.46             -666            -66.6   
...                 ...              ...              ...              ...   
127015            -66.6             0.44             -666            -66.6   
127016            -66.6             0.43             -666            -66.6   
127017            -66.6             0.43             -666            -66.6   
127018            -66.6             0.43             -666            -66.6   
127019            -66.6             0.44             -666            -66.6   

        MTD-PRCP-NORMAL  MTD-SNOW-NORMAL  YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  \
0                  0.04              0.0             0.04              0.0   
1                  0.08              0.0             0.08              0.0   
2                  0.12              0.0             0.12              0.0   
3                  0.16              0.0             0.16              0.0   
4                  0.21              0.0             0.21              0.0   
...                 ...              ...              ...              ...   
127015             1.14              0.3            11.29              1.5   
127016             1.19              0.3            11.34              1.5   
127017             1.24              0.3            11.39              1.5   
127018             1.30              0.3            11.45              1.5   
127019             1.35              0.3            11.50              1.5   

        DLY-PRCP-PCTALL-GE001HI  DLY-PRCP-PCTALL-GE010HI  \
0                           137                       81   
1                           138                       83   
2                           139                       84   
3   

In [72]:
# set the DATE and STATION as index

a = prep_df.set_index(['DATE', 'STATION'])
b = temp_df.set_index(['DATE', 'STATION'])

In [73]:
a.head()

STATION_NAME  DLY-PRCP-25PCTL  \
DATE     STATION                                                      
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   

                            DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   

                            DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   

                            DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6             0.02   
20100102 GHCND:USC00049099             -666            -66.6             0.04   
20100103 GHCND:USC00049099             -666            -66.6             0.05   
20100104 GHCND:USC00049099            -9999          -9999.0             0.07   
20100105 GHCND:USC00049099            -9999          -9999.0             0.09   

                            MTD-SNOW-NORMAL  YTD-PRCP-NORMAL  YTD-SNOW-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099              0.0             0.02              0.0   
20100102 GHCND:USC00049099              0.0             0.04              0.0   
20100103 GHCND:USC00049099              0.0             0.05              0.0   
20100104 GHCND:USC00049099              0.0             0.07              0.0   
20100105 GHCND:USC00049099              0.0             0.09              0.0   

                            DLY-PRCP-PCTALL-GE001HI  DLY-PRCP-PCTALL-GE010HI  \
DATE     STATION                                                               
20100101 GHCND:USC00049099                       98                       43   
20100102 GHCND:USC00049099                       99                       44   
20100103 GHCND:USC00049099                      100                       44   
20100104 GHCND:USC00049099                      101                       45   
20100105 GHCND:USC00049099                      102                       45   

                            DLY-PRCP-PCTALL-GE050HI  DLY-PRCP-PCTALL-GE100HI  \
DATE     STATION                                                               
20100101 GHCND:USC00049099                       12                        3   
20100102 GHCND:USC00049099                       12                        3   
20100103 GHCND:USC00049099                       12                        3   
20100104 GHCND:USC00049099                       12                        3   
20100105 GHCND:USC00049099                       12                        3   

                            DLY-SNWD-PCTALL-GE001WI  DLY-SNWD-PCTALL-GE010WI  \
DATE     STATION                                                  

In [74]:
b.head()

STATION_NAME ELEVATION  LATITUDE  \
DATE     STATION                                                         
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US       602  34.12806   

                             LONGITUDE  DLY-CLDD-BASE45  DLY-CLDD-BASE50  \
DATE     STATION                                                           
20100101 GHCND:USC00049099  -116.03694                6                2   
20100102 GHCND:USC00049099  -116.03694                6                2   
20100103 GHCND:USC00049099  -116.03694                6                2   
20100104 GHCND:USC00049099  -116.03694                6                2   
20100105 GHCND:USC00049099  -116.03694                6                2   

                            DLY-CLDD-BASE55  DLY-CLDD-BASE57  DLY-CLDD-BASE60  \
DATE     STATION                                                                
20100101 GHCND:USC00049099            -7777            -7777            -7777   
20100102 GHCND:USC00049099                1            -7777            -7777   
20100103 GHCND:USC00049099                1            -7777            -7777   
20100104 GHCND:USC00049099                1            -7777            -7777   
20100105 GHCND:USC00049099                1            -7777            -7777   

                            DLY-CLDD-NORMAL  DLY-CLDD-BASE70  DLY-CLDD-BASE72  \
DATE     STATION                                                                
20100101 GHCND:USC00049099                0                0                0   
20100102 GHCND:USC00049099                0                0                0   
20100103 GHCND:USC00049099                0                0                0   
20100104 GHCND:USC00049099                0                0                0   
20100105 GHCND:USC00049099                0                0                0   

                            DLY-HTDD-BASE40  DLY-HTDD-BASE45  DLY-HTDD-BASE50  \
DATE     STATION                                                                
20100101 GHCND:USC00049099            -7777                1                2   
20100102 GHCND:USC00049099            -7777                1                2   
20100103 GHCND:USC00049099            -7777                1                2   
20100104 GHCND:USC00049099            -7777                1                2   
20100105 GHCND:USC00049099            -7777            -7777                2   

                            DLY-HTDD-BASE55  DLY-HTDD-BASE57  DLY-HTDD-BASE60  \
DATE     STATION                                                                
20100101 GHCND:USC00049099                6                7               10   
20100102 GHCND:USC00049099                6                7               10   
20100103 GHCND:USC00049099                5                7               10   
20100104 GHCND:USC00049099                5                7               10   
20100105 GHCND:USC00049099                5                7               10   

                            DLY-HTDD-NORMAL  
DATE     STATION                             
20100101 GHCND:USC00049099               15  
20100102 GHCND:USC00049099               15  
20100103 GHCND:USC00049099               15  
20100104 GHCND:USC00049099               15  
20100105 GHCND:USC00049099               15

In [75]:
a.join(b)

ValueError: columns overlap but no suffix specified: Index(['STATION_NAME'], dtype='object')

In [76]:
a.join(b, lsuffix='_left', rsuffix='_right')

STATION_NAME_left  DLY-PRCP-25PCTL  \
DATE     STATION                                                      
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
...                                            ...              ...   
20101227 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101228 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101229 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101230 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101231 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   

                            DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             0.41   
20101228 GHCND:USC00046006            -9999          -9999.0             0.42   
20101229 GHCND:USC00046006            -9999          -9999.0             0.43   
20101230 GHCND:USC00046006            -9999          -9999.0             0.43   
20101231 GHCND:USC00046006            -9999          -9999.0             0.44   

                            DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             1.13   
20101228 GHCND:USC00046006            -9999          -9999.0             1.14   
20101229 GHCND:USC00046006            -9999          -9999.0             1.15   
20101230 GHCND:USC00046006            -9999          -9999.0             1.15   
20101231 GHCND:USC00046006            -9999          -9999.0             1.16   

                            DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6             0.02   
20100102 GHCND:USC00049099             -666            -66.6             0.04   
20100103 GHCND:USC00049099             -666            -66.6             0.05   
20100104 GHCND:USC00049099            -9999          -9999.0             0.07   
20100105 GHCND:USC00049099            -9999          -9999.0             0.09   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             3.91   
20101228 GHCND:USC00046006            -9999          -9999.0             4.10   
20101229 GHCND:USC00046006            -9999          -9999.0             4.30   
20101230 GHCND:USC00046006            -9999          -9999.0             4.50   
20101231 GHCND:USC00046006   

In [77]:
# Concatenating the temp_df with itself along row-wise
pd.concat([a, b], axis=0)   

STATION_NAME  DLY-PRCP-25PCTL  \
DATE     STATION                                                      
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
...                                            ...              ...   
20101227 GHCND:USC00046006  MOUNT WILSON CBS CA US              NaN   
20101228 GHCND:USC00046006  MOUNT WILSON CBS CA US              NaN   
20101229 GHCND:USC00046006  MOUNT WILSON CBS CA US              NaN   
20101230 GHCND:USC00046006  MOUNT WILSON CBS CA US              NaN   
20101231 GHCND:USC00046006  MOUNT WILSON CBS CA US              NaN   

                            DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100102 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100103 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100104 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
20100105 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006              NaN              NaN              NaN   
20101228 GHCND:USC00046006              NaN              NaN              NaN   
20101229 GHCND:USC00046006              NaN              NaN              NaN   
20101230 GHCND:USC00046006              NaN              NaN              NaN   
20101231 GHCND:USC00046006              NaN              NaN              NaN   

                            DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100102 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100103 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100104 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
20100105 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006              NaN              NaN              NaN   
20101228 GHCND:USC00046006              NaN              NaN              NaN   
20101229 GHCND:USC00046006              NaN              NaN              NaN   
20101230 GHCND:USC00046006              NaN              NaN              NaN   
20101231 GHCND:USC00046006              NaN              NaN              NaN   

                            DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6             0.02   
20100102 GHCND:USC00049099           -666.0            -66.6             0.04   
20100103 GHCND:USC00049099           -666.0            -66.6             0.05   
20100104 GHCND:USC00049099          -9999.0          -9999.0             0.07   
20100105 GHCND:USC00049099          -9999.0          -9999.0             0.09   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006              NaN              NaN              NaN   
20101228 GHCND:USC00046006              NaN              NaN              NaN   
20101229 GHCND:USC00046006              NaN              NaN              NaN   
20101230 GHCND:USC00046006              NaN              NaN              NaN   
20101231 GHCND:USC00046006        

In [81]:
# Concatenating the teperature dataset with precipitation dataset along column-wise
# pd.concat([prep_df, temp_df], axis=1)
pd.concat([a, b], axis=1)

STATION_NAME  DLY-PRCP-25PCTL  \
DATE     STATION                                                      
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
...                                            ...              ...   
20101227 GHCND:USC00045280                     NaN              NaN   
20101228 GHCND:USC00045280                     NaN              NaN   
20101229 GHCND:USC00045280                     NaN              NaN   
20101230 GHCND:USC00045280                     NaN              NaN   
20101231 GHCND:USC00045280                     NaN              NaN   

                            DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100102 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100103 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100104 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
20100105 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00045280              NaN              NaN              NaN   
20101228 GHCND:USC00045280              NaN              NaN              NaN   
20101229 GHCND:USC00045280              NaN              NaN              NaN   
20101230 GHCND:USC00045280              NaN              NaN              NaN   
20101231 GHCND:USC00045280              NaN              NaN              NaN   

                            DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100102 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100103 GHCND:USC00049099           -666.0            -66.6            -6.66   
20100104 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
20100105 GHCND:USC00049099          -9999.0          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00045280              NaN              NaN              NaN   
20101228 GHCND:USC00045280              NaN              NaN              NaN   
20101229 GHCND:USC00045280              NaN              NaN              NaN   
20101230 GHCND:USC00045280              NaN              NaN              NaN   
20101231 GHCND:USC00045280              NaN              NaN              NaN   

                            DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099           -666.0            -66.6             0.02   
20100102 GHCND:USC00049099           -666.0            -66.6             0.04   
20100103 GHCND:USC00049099           -666.0            -66.6             0.05   
20100104 GHCND:USC00049099          -9999.0          -9999.0             0.07   
20100105 GHCND:USC00049099          -9999.0          -9999.0             0.09   
...                                     ...              ...              ...   
20101227 GHCND:USC00045280              NaN              NaN              NaN   
20101228 GHCND:USC00045280              NaN              NaN              NaN   
20101229 GHCND:USC00045280              NaN              NaN              NaN   
20101230 GHCND:USC00045280              NaN              NaN              NaN   
20101231 GHCND:USC00045280        

In [84]:
# pd.concat([prep_df, temp_df], axis=1, join='inner')
pd.concat([a, b], axis=1, join='inner')

STATION_NAME  DLY-PRCP-25PCTL  \
DATE     STATION                                                      
20100101 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100102 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100103 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100104 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
20100105 GHCND:USC00049099  TWENTYNINE PALMS CA US            -6.66   
...                                            ...              ...   
20101227 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101228 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101229 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101230 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   
20101231 GHCND:USC00046006  MOUNT WILSON CBS CA US             0.12   

                            DLY-SNWD-25PCTL  DLY-SNOW-25PCTL  DLY-PRCP-50PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             0.41   
20101228 GHCND:USC00046006            -9999          -9999.0             0.42   
20101229 GHCND:USC00046006            -9999          -9999.0             0.43   
20101230 GHCND:USC00046006            -9999          -9999.0             0.43   
20101231 GHCND:USC00046006            -9999          -9999.0             0.44   

                            DLY-SNWD-50PCTL  DLY-SNOW-50PCTL  DLY-PRCP-75PCTL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6            -6.66   
20100102 GHCND:USC00049099             -666            -66.6            -6.66   
20100103 GHCND:USC00049099             -666            -66.6            -6.66   
20100104 GHCND:USC00049099            -9999          -9999.0            -6.66   
20100105 GHCND:USC00049099            -9999          -9999.0            -6.66   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             1.13   
20101228 GHCND:USC00046006            -9999          -9999.0             1.14   
20101229 GHCND:USC00046006            -9999          -9999.0             1.15   
20101230 GHCND:USC00046006            -9999          -9999.0             1.15   
20101231 GHCND:USC00046006            -9999          -9999.0             1.16   

                            DLY-SNWD-75PCTL  DLY-SNOW-75PCTL  MTD-PRCP-NORMAL  \
DATE     STATION                                                                
20100101 GHCND:USC00049099             -666            -66.6             0.02   
20100102 GHCND:USC00049099             -666            -66.6             0.04   
20100103 GHCND:USC00049099             -666            -66.6             0.05   
20100104 GHCND:USC00049099            -9999          -9999.0             0.07   
20100105 GHCND:USC00049099            -9999          -9999.0             0.09   
...                                     ...              ...              ...   
20101227 GHCND:USC00046006            -9999          -9999.0             3.91   
20101228 GHCND:USC00046006            -9999          -9999.0             4.10   
20101229 GHCND:USC00046006            -9999          -9999.0             4.30   
20101230 GHCND:USC00046006            -9999          -9999.0             4.50   
20101231 GHCND:USC00046006        

## The dimension of the data is too vast and i cant decide which features should i take for EDA purpose.

In [86]:
prep_df.isnull().sum().sum(), temp_df.isnull().sum().sum()

(0, 0)

There is no missing values in the provided datasets so no need to apply imputation.

In [87]:
# Save the processed data
# prep_df.to_csv('processed_climate_precip.csv', index=False)
# temp_df.to_csv('processed_climate_temp.csv', index=False)